
syntax = "proto3";

package test;

// Interface exported by the server.
service Control 
{
    rpc Shutdown (ShutdownRequest) returns (ShutdownResponse) {}
    rpc Status (StatusRequest) returns (StatusResponse) {}
}

message ShutdownRequest
{
    string reason = 1;
}

message ShutdownResponse
{
    bool result = 1;
    string msg = 2;
}

message StatusRequest
{
    string service_name = 1;
}

message StatusResponse
{
    string service_status = 1;
}

